### PR TITLE
 Refactored types in frame to use size_t (#fix 34)

### PR DESCRIPTION
**Pull request by o3o**
It compiles to ARM, but
I have a doubt about line 124.
```
writeDataOrThrow(ptr, cast(ulong) frame.payload.length);
```

perhaps it should be changed to `cast(size_t)`?

### CODE REVIEW

File: source/handy_httpd/components/websocket/frame.d [hunk 1]
```diff
@@ -223,11 +223,11 @@ private bool validateOpcode(ubyte opcode) {
  *   stream = The stream to read from.
  * Returns: The complete payload length.
  */
-private ulong readPayloadLength(S)(ubyte initialLength, S stream) if (isByteInputStream!S) {
+private size_t readPayloadLength(S)(ubyte initialLength, S stream) if (isByteInputStream!S) {
     if (initialLength == 126) {
         return readDataOrThrow!(ushort)(stream);
     } else if (initialLength == 127) {
-        return readDataOrThrow!(ulong)(stream);
+        return readDataOrThrow!(size_t)(stream);
```
Comment (andrewlalis) [inline source/handy_httpd/components/websocket/frame.d:230]: The specification requires that the payload length be encoded as 64 bits if the initial length is 127, so it's gotta stay as `readDataOrThrow!(ulong)(stream)`. I guess the best thing to do would be `return cast(size_t) readDataOrThrow!(ulong)(stream);`

**Review by andrewlalis — CHANGES_REQUESTED** [review]

**Comment by andrewlalis** [comment]
Regarding your question about line 124, the `writeDataOrThrow` call there does **need** to write a ulong, since the spec requires 64 bits in that part of the frame. ARM only seems to complain when a long or ulong is used instead of size_t when doing array indexing operations.

**Comment by andrewlalis** [comment]
Also, referencing #34 so that GitHub knows the issue is linked to this pr.

File: source/handy_httpd/components/websocket/frame.d (hunk 1 above)
Comment (o3o) [reply source/handy_httpd/components/websocket/frame.d:230]: you are right, the frame length could be [64bit](https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API/Writing_WebSocket_servers#decoding_payload_length),  but if I use ulong in 32 arm the compiler complains. I could do a cast from ulong to uint, but if actually the package is bigger than uint it generates a conversion error.... I have to think more about it. Thanks for the great support

**Comment by andrewlalis** [comment]
Closing this because I implemented your changes, plus the details I mentioned in commit 4a22ed4

**Pull request closed without merging**